nautobot/core/celery/__init__.py

```diff
@@ -25,7 +28,22 @@
 # NOT need to be called here.
 # nautobot.setup()
 
-app = Celery("nautobot", task_cls="nautobot.core.celery.task:NautobotTask")
+
+class NautobotCelery(Celery):
+    task_cls = "nautobot.core.celery.task:NautobotTask"
+
+    def register_task(self, task, **options):
+        """Override the default task name for job classes to allow app provided jobs to use the full module path."""
+        from nautobot.extras.jobs import Job
```

Comment:
Should now be a time to move the `Job` class (not the `Job` database model) from `extras` to `core`? Would that simplify things?